### ISSUE TITLE
The road to 1.0

### ISSUE THREAD

**Issue by wastaz** [wastaz/Hyperboliq#8]
The library is in a pretty stable-ish state, but there are a couple of tasks that needs to be done before I feel totally safe with releasing this as 1.0. 

Must-haves for 1.0
- [ ] Split CSharp and FSharp API's into separate nuget packages `Hyperboliq.CSharp` and `Hyperboliq.FSharp`. The compiler can currently get slightly confused occasionally when these two api's are together as they currently are, and I see no reason why you would want to have one if you are using the other.
- [ ] Remove the custom data mapping part. For me personally this library is mainly about the query generation, other libraries do the data mapping much better than I could ever do. And as such, we should not provide the actual data mapping parts. Add Dapper and integrate that properly instead, no need to invent the wheel twice.

Nice-to-haves for 1.0
- [ ] Investigate if it is worth it to switch test frameworks again to Expecto. Expecto seems to be much nicer than NUnit and I can easily see myself more happy with using it in the future.
- [ ] Documentation. While the current README contains a decent enough amount of examples it would certainly be nice to have some sort of nicer docs to point to.
- [ ] Normalize coding style. In the middle of the project I seem to have switched coding style. While this is not

**Comment by wallymathieu** [comment]
Would it be possible to combine with entity framework core for the data mapping?
